rhp/v4/server.go

```diff
@@ -585,14 +585,14 @@ func (s *Server) handleRPCFormContract(stream net.Conn) error {
 
 	// update renter input basis to reflect our funding basis
 	if basis != req.Basis {
-		hostInputs := formationTxn.SiacoinInputs[len(formationTxn.SiacoinInputs)-len(req.RenterInputs)]
-		formationTxn.SiacoinInputs = formationTxn.SiacoinInputs[:len(formationTxn.SiacoinInputs)-len(req.RenterInputs)]
+		hostInputs := formationTxn.SiacoinInputs[len(req.RenterInputs):]
+		formationTxn.SiacoinInputs = formationTxn.SiacoinInputs[:len(req.RenterInputs)]
 		txnset, err := s.chain.UpdateV2TransactionSet([]types.V2Transaction{formationTxn}, req.Basis, basis)
 		if err != nil {
 			return errorBadRequest("failed to update renter inputs from %q to %q: %v", req.Basis, basis, err)
 		}
 		formationTxn = txnset[0]
-		formationTxn.SiacoinInputs = append(formationTxn.SiacoinInputs, hostInputs)
+		formationTxn.SiacoinInputs = append(formationTxn.SiacoinInputs, hostInputs...)
```

Comment:
what happens if we call `UpdateV2TransactionSet` only all the inputs, i.e. without modifying `formationTxn`?

Reply:
Is it safe to call update on a transaction where some of the inputs are already updated?

Reply:
hmm, yeah probably not